frontend/awx/resources/templates/WorkflowVisualizer/hooks/useWorkflowVisualizerToolbarActions.tsx

```diff
@@ -5,19 +5,71 @@ import {
   CheckCircleIcon,
   CloseIcon,
   CompressAltIcon,
+  EllipsisVIcon,
   ExpandAltIcon,
   PlusCircleIcon,
+  TrashIcon,
 } from '@patternfly/react-icons';
-import { Badge, Button, ToolbarGroup, ToolbarItem } from '@patternfly/react-core';
+import {
+  Badge,
+  Button,
+  Dropdown,
+  DropdownItem,
+  DropdownList,
+  MenuToggle,
+  ToolbarGroup,
+  ToolbarItem,
+} from '@patternfly/react-core';
 import { WorkflowNode } from '../../../../interfaces/WorkflowNode';
-import { usePageNavigate } from '../../../../../../framework';
+import { useBulkConfirmation, usePageNavigate } from '../../../../../../framework';
 import { AwxRoute } from '../../../../AwxRoutes';
+import { getItemKey } from '../../../../../common/crud/Data';
+import { stringIsUUID } from '../../../../common/util/strings';
 
 export function useWorkflowVisualizerToolbarActions(nodes: WorkflowNode[]) {
   const { t } = useTranslation();
   const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [isKebabOpen, setIsKebabOpen] = useState<boolean>(false);
   const pageNavigate = usePageNavigate();
   const params = useParams<{ id: string }>();
+  const bulkAction = useBulkConfirmation<WorkflowNode>();
+  const handleRemoveAllNodes = useCallback(
+    (workflowNodes: WorkflowNode[]) => {
+      return bulkAction({
+        title: t('Remove all nodes'),
+        items: workflowNodes,
+        keyFn: getItemKey,
+        isDanger: true,
+        actionFn: async (_item: WorkflowNode, _signal: AbortSignal) => {
+          // Add your code here
```

Comment:
Chatted with @AlexSCorey and we decided to pull in the "Remove all nodes" button into the toolbar. However, we will disable its functionality as we are in the process of finding the best approach for managing node states.